cEP-0020.md

```diff
@@ -0,0 +1,501 @@
+
+To increase a user's competitive instinct, we can implement a
+series of levels that confer rank as newcomers become more
+active. A good and straightforward way to award levels in a
+new gamified design is to base them on point thresholds. As
+users earn points, they move up an incremental series of levels.
+To infuse the level with meaning i.e.: a ranking within the
+system, the levels should be named in a way that indicates status.
+
+But the levels wouldn't be enough motivation to go until newcomers
+get some special privileges after completing each level:
+
+- Level-I:
+  - Name: Fresher
+  - Points_required = 0
+  - Privileges:
+    - No privileges for freshers
+- Level-II:
+  - Name: Newbie
+  - Points_required = 10
+  - Privileges:
+    - Get assigned to newcomers issue
+- Level-III:
+  - Name: Beginner
+  - Points_required = 20
+  - Privileges:
+    - Get assigned to a difficulty low issue
+- Level-IV:
+  - Name: Intermediate
+  - Points_required = 50
+  - Privileges:
+    - Mark your own PR as wip
+- Level-V:
+  - Name: Cool
+  - Points_required = 100
+  - Privileges:
+    - Mark a PR by another newcomer as wip
+    - Get assigned to difficulty medium issue
+- Level-VII:
+  - Name: Expert
+  - Points_required = 200
+  - Privileges
+    - Assign newcomers to difficulty newcomer issue
+    - Assign newcomers to difficulty low issue
+- Level-VIII:
+  - Name: Legend
+  - Points_required = 350
+  - Privileges
+    - Invite newcomers to org
+    - Assign newcomers to difficulty medium issue
+    - Permission to `ack` and `unack` on the pull requests
+
+Implementing the above privileges is beyond the scope of
+the gamification system.
+The gamification system will merely expose a list of privileges
+that each person has, and other systems i.e.:
+[corobo](https://github.com/coala/corobo)
+are expected to adapt to use these privileges.
+
+While implementing this process, a first manual check should
+be done for testing purpose, and then automated only if there
+is no serious risk or there is a good risk reaction mechanism
+in place.
+
+### Badges
+
+Another way of explicitly nudging a newcomer to action is to
+award badges for completing tasks. Badges enable newcomers to
+follow their performance and compare to other newcomers.
+Badges will be awarded after performing certain activities by
+the newcomers:
+
+- Badge-I:
+  - Name: The bug finder
+  - Details: The one who find bugs in the existing codebase
+  - Activities:
+    - Create at least one issue which labeled as bug
+- Badge-II:
+  - Name: The bear hunter
+  - Details: The one who create issues about new bears
+  - Activities:
+    - Create at least one bear proposal issue
+- Badge-III:
+  - Name: The bear writer
+  - Details: The one who write new bears
+  - Activities:
+    - Merged at least 2 bear proposal PRs
+- Badge-IV:
+  - Name: The helper
+  - Details: The one who answers every newcomer questions on Gitter
+  - Activities:
+    - At least 20 good messages to newcomer
```

Comment:
The idea behind awarding badges according to a single type of activity is to show people that which newcomer is best in performing a certain type of tasks, which can also be useful outside the coala community, in case a newcomer is not able to become a developer at coala.

Reply:
I agree with your goal there, but you can have many examples here achieving your goal and fix the problem I mentioned above.